kubernetes/Upgrading_CoreDNS.md

```diff
@@ -0,0 +1,199 @@
+
+# Handling CoreDNS Upgrades in Kubernetes Clusters
+
+When upgrading a Kubernetes cluster, some care should be taken when upgrading the CoreDNS component to avoid backward incompatible configuration failures. At a high level, you'll want to review the CoreDNS release notes to see if the new version introduces backward incompatibilities or deprecations and adjust your Corefile as needed before upgrading CoreDNS.
+
+
+## Identifying Backward Incompatibilities and Deprecations
+
+If you are upgrading CoreDNS as part of a Kubernetes upgrade, backward incompatibilities should be noted in the Kubernetes release notes of each version. You may also review the CoreDNS release notes as described below.
```

Comment:
Are there any backward incompatible changes mentioned in the Kubernetes releases yet? 

K8s release notes https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md
Maybe we can add this link as well.

So we'll need to mention backward incompatible changes on the Kubernetes side while writing the release notes for PRs updating the CoreDNS versions.

Reply:
It means we would have to maintain it, but i think it's a good idea.

If we want to do that, I think it should be in a separate document.

Reply:
sgtm, I will start a new doc on this. 
Maybe link that to this document later.